### ISSUE TITLE
Patch -> SaveChanges -> Load: stale result

### ISSUE THREAD

**Issue by migajek** [JasperFx/marten#925]
When modifying document(s) via patch api, calling `SaveChanges` and then `Load` does not work as expected - returns stale result. See tests below.

I know one could manually `Eject` the document, but then again it'd be required to `Load` it first.

For me, this breaks the scenario of events projection - in reaction to one event, the documents are patched. Reacting to other events, documents are Loaded first and then some data is copied from one document to another. All of this happens in one session (managed by async daemon)

One fix that came to my mind is to eject all documents matching the Patch'es `IWhereFragment` when applying it. What do you think of this approach? I think I could try implementing it

tests: 
```
        [Fact]
        public async Task Patch()
        {
            var id = Guid.NewGuid();
            using (var sess = theStore.OpenSession())
            {
                sess.Store(new Model() { Id = id, Name = "foo" });
                sess.Patch<Model>(id).Set(x => x.Name, "bar");
                await sess.SaveChangesAsync();                
                sess.Query<Model>().Where(x => x.Id == id).Select(x => x.Name).Single().ShouldBe("bar");
                sess.Load<Model>(id).Name.ShouldBe("bar"); // fails. Name is null.
            }
        }

        [Fact] // this one passes
        public async Task LoadAndChange()
        {
            var id = Guid.NewGuid();
            using (var sess = theStore.OpenSession())
            {
                sess.Store(new Model(){Id = id, Name = "foo"});
                sess.Load<Model>(id).Name = "bar";
                await sess.SaveChangesAsync();                
                sess.Query<Model>().Where(x => x.Id == id).Select(x => x.Name).Single().ShouldBe("bar");
                sess.Load<Model>(id).Name.ShouldBe("bar");
            }
        }
```

**Comment by jeremydmiller** [comment]
Ugh, that's going to be annoying. If there's a patch, I'd vote to just eject all models of that type. Might end up being cheaper at runtime than it would be to do it selectively

**Comment by migajek** [comment]
ok, this sounds to be even easier. I'll try a PR

**Comment by jeremydmiller** [comment]
Fixed by #926
